src/sqlfluff/core/rules/std/L015.py

```diff
@@ -30,10 +30,42 @@ def _eval(self, segment, raw_stack, **kwargs):
 
         Look for DISTINCT keyword immediately followed by open parenthesis.
         """
-        # We only trigger on start_bracket (open parenthesis)
-        if segment.name == "start_bracket":
-            filt_raw_stack = self.filter_meta(raw_stack)
-            if len(filt_raw_stack) > 0 and filt_raw_stack[-1].name == "DISTINCT":
-                # If we find DISTINCT followed by open_bracket, then bad.
-                return LintResult(anchor=segment)
-        return LintResult()
+        # We only trigger on expressions that begin with start_bracket (open
+        # parenthesis)
+        if segment.type == "expression":
+            segments_filt = self.filter_meta(segment.segments)
+            if (
+                len(segments_filt)
```

Comment:
```suggestion
                segments_filtered
```

Don't think you need len here

Reply:
Agreed. I'll remove it. My first language was C, and I still prefer to be explicit about things like this, even though it's not the Python way. 😄